src/main/java/de/uni_passau/fim/se2/pipeline_helper/checkers/line_length/FileLineLengthViolations.java

```diff
@@ -0,0 +1,25 @@
+// SPDX-FileCopyrightText: 2022 Pipeline Helper Contributors
+//
+// SPDX-License-Identifier: EUPL-1.2
+
+package de.uni_passau.fim.se2.pipeline_helper.checkers.line_length;
+
+import java.nio.file.Path;
+import java.util.List;
+
+public record FileLineLengthViolations(Path file, int count, List<Integer> lines) {
+    @Override
+    public String toString() {
```

Comment:
This breaks in cases

- single violation in a file: `Board.java, on 1 lines & 3`
  - plural `s` is okay, but `&`
- when the last line number has more than one digit: `Board.java, on 2 lines: 3,  & 1`
  - actually line 152

I’d recommend using `String.join(", ", lines.subList(...))` to comma-separate all but the last element, and then adding that last item at the end manually rather than trying to remove from the built string.

Please also add additional tests for those cases.

Reply:
I didn't even know about the join function! Again what learned.

Reply:
Second case is now already covered by the existing tests, since incrementing the line numbers by one turns the 9 into 10.